2.0/m_immunemode.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2014 Miguel Peláez <miguel2706@outlook.com>
+ *   Copyright (C) 2014-2015 Hira.io Team <staff@hira.io>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/* $ModAuthor: Hira.io Team */
+/* $ModDesc: Module to make an operator immune for a specified time, it's like a uline */
+/* $ModDepends: core 2.0 */
+
+#include "inspircd.h"
+int Timeout = 0;
+struct AutoRemoveMode : public Timer
+{
+ public:
+        std::string userid;
+        AutoRemoveMode(long interval)
+                : Timer(interval, ServerInstance->Time(), false)
+        {
+        }
+        void Tick(time_t TIME)
+        {
+		User* user = ServerInstance->FindUUID(userid);
+		if(!user)
+			return;
+		user->WriteServ("NOTICE "+user->nick+" :*** Removing your user mode +t to prevent abuses.");
+		std::vector<std::string> modes;
+		modes.push_back(user->nick);
+		modes.push_back("-t");
+		ServerInstance->SendGlobalMode(modes, ServerInstance->FakeClient);
+        }
+};
+/** User mode +t - Immune from kicks
+ */
+class ImmuneMode : public SimpleUserModeHandler
+{
+ public:
+	ImmuneMode(Module* Creator) : SimpleUserModeHandler(Creator, "immune", 't')
+	{
+		oper = true;
+	}
+	ModeAction OnModeChange(User* source, User* dest, Channel* channel, std::string &parameter, bool adding)
+        {
+		if(!adding)
+			return MODEACTION_ALLOW;
+		if(!IS_LOCAL(dest))
+			return MODEACTION_ALLOW;
+		if((!Timeout)||(Timeout == 0))
+		{
+			dest->WriteServ("NOTICE "+dest->nick+" :*** This mode will be in place indefinitely, remember to take it off to itself to prevent abuse of power or setting the automatic removal.");
+			return MODEACTION_ALLOW;
+		}
+                AutoRemoveMode *timer = new AutoRemoveMode(Timeout);
```

Comment:
You should probably store a list over active timers so you can delete them on module unload.